From 8c42e0c85b8be5793ecf67c9ae1123f1343c5bed Mon Sep 17 00:00:00 2001
From: John Reck <jreck@google.com>
Date: Wed, 25 Jun 2025 14:18:45 -0700
Subject: [PATCH] Track DNG SDK 1.7.1 API changes

966e57a9a04b3dc56de8230e4c02d73ac46c01b2

Bug: 412662901
Test: make
Cherrypick-From: https://googleplex-android-review.googlesource.com/q/commit:710c53b4c71495ba646646ba11510f423ae458e7
Merged-In: I11050ac5461b13a65fd6d69b4607ffc9ab3fba2c
Change-Id: I11050ac5461b13a65fd6d69b4607ffc9ab3fba2c
---
 .../libctscamera2jni/dng-validate-jni.cpp     | 41 +++++++++----------
 1 file changed, 20 insertions(+), 21 deletions(-)

diff --git a/tests/camera/libctscamera2jni/dng-validate-jni.cpp b/tests/camera/libctscamera2jni/dng-validate-jni.cpp
index 186cf3f3b97..e6bb3aab406 100644
--- a/tests/camera/libctscamera2jni/dng-validate-jni.cpp
+++ b/tests/camera/libctscamera2jni/dng-validate-jni.cpp
@@ -234,7 +234,7 @@ static dng_error_code dng_validate(const void* data, uint32_t count) {
 
                 // Skip preview if writing a compresssed main image to save space
                 // in this example code.
-                if (negative->RawJPEGImage() != NULL && previewIndex > 0) {
+                if (negative->RawLossyCompressedImage() != NULL && previewIndex > 0) {
                     break;
                 }
 
@@ -265,8 +265,8 @@ static dng_error_code dng_validate(const void* data, uint32_t count) {
 
                 // If we have compressed JPEG data, create a compressed thumbnail.  Otherwise
                 // save a uncompressed thumbnail.
-                bool useCompressedPreview = (negative->RawJPEGImage() != NULL) ||
-                        (previewIndex > 0);
+                bool useCompressedPreview =
+                        (negative->RawLossyCompressedImage() != NULL) || (previewIndex > 0);
 
                 AutoPtr<dng_preview> preview (useCompressedPreview ?
                         (dng_preview *) new dng_jpeg_preview :
@@ -287,7 +287,7 @@ static dng_error_code dng_validate(const void* data, uint32_t count) {
 
                 if (!useCompressedPreview) {
                     dng_image_preview *imagePreview = static_cast<dng_image_preview *>(preview.Get());
-                    imagePreview->fImage.Reset(previewImage.Release());
+                    imagePreview->SetImage(host, previewImage.Release());
                 } else {
                     dng_jpeg_preview *jpegPreview = static_cast<dng_jpeg_preview *>(preview.Get());
                     int32 quality = (previewIndex == 0 ? 8 : 5);
@@ -308,12 +308,8 @@ static dng_error_code dng_validate(const void* data, uint32_t count) {
                 dng_timer timer("Write DNG time");
                 dng_image_writer writer;
 
-                writer.WriteDNG(host,
-                        stream2,
-                        *negative.Get(),
-                        &previewList,
-                        dngVersion_Current,
-                        false);
+                writer.WriteDNG(host, stream2, *negative.Get(), &previewList,
+                                dngVersion_SaveDefault, false);
             }
 
             gDumpDNG.Clear();
@@ -351,6 +347,8 @@ static dng_error_code dng_validate(const void* data, uint32_t count) {
             if (negative->GetXMP()) {
                 negative->GetXMP()->RemoveProperties(XMP_NS_CRS);
                 negative->GetXMP()->RemoveProperties(XMP_NS_CRSS);
+                negative->GetXMP()->RemoveProperties(XMP_NS_CRD);
+                negative->GetXMP()->RemoveProperties(XMP_NS_CRLCP);
             }
 #endif
 
@@ -358,17 +356,18 @@ static dng_error_code dng_validate(const void* data, uint32_t count) {
             dng_file_stream stream2(gDumpTIF.Get(), true);
 
             {
+                const dng_camera_profile *profilePtr = nullptr;
+                dng_camera_profile profile;
+                if (!negative->IsMonochrome()) {
+                    const auto &profileID = render.CameraProfileID();
+                    if (negative->GetProfileByID(profileID, profile)) profilePtr = &profile;
+                }
                 dng_timer timer("Write TIFF time");
                 dng_image_writer writer;
 
-                writer.WriteTIFF(host,
-                        stream2,
-                        *finalImage.Get(),
-                        finalImage->Planes() >= 3 ? piRGB
-                        : piBlackIsZero,
-                        ccUncompressed,
-                        negative.Get(),
-                        &render.FinalSpace());
+                writer.WriteTIFF(host, stream2, *finalImage.Get(),
+                                 finalImage->Planes() >= 3 ? piRGB : piBlackIsZero, ccUncompressed,
+                                 &negative->Metadata(), &render.FinalSpace(profilePtr));
             }
             gDumpTIF.Clear();
         }
@@ -432,7 +431,7 @@ Java_android_hardware_camera2_cts_DngCreatorTest_validateDngNative(
         ALOGE("Error reading from dng_validate output pipe: %d", errno);
         return JNI_FALSE;
     }
-    close(pipeFds[1]);
+    close(pipeFds[0]);
 
     std::string line;
     int lineCount = 0;
@@ -457,8 +456,8 @@ Java_android_hardware_camera2_cts_DngCreatorTest_validateDngNative(
         }
     }
     // If no output is produced, assume something went wrong
-    if (lineCount < 3) {
-        ALOGE("Validation output less than expected!");
+    if (lineCount > 0) {
+        ALOGE("Unexpected stderr output!");
         dng_err = dng_error_unknown;
     }
     if (dng_err != dng_error_none) {
-- 
2.52.0.rc1.455.g30608eb744-goog

